environment_unix.yml

```
@@ -68,9 +68,9 @@ dependencies:
   # See: https://github.com/man-group/ArcticDB/pull/291
   - hypothesis < 6.73
   - pytest-sugar
-  - pymongo
   # Python dependencies (for tests only)
   - azure-storage-blob
   - azure-identity
   - pyarrow
   - asv
+  - pymongo
```

Shall we drop pymongo here? Bit dodgy but then at least we'll have PyPi CI run _with_ pymongo and Conda run _without_, so we can cover both code paths

---

Wow that's hacky genius thought. Good idea.

---

I have come up with a better idea on the testing, with parametrised pytest + monkeypatch

---

[`environment_unix.yml`](https://github.com/man-group/ArcticDB/pull/1027/files/8c05b51c36ab79700ea0bbc80373f020eb91b921#diff-f557abdd91aaee31705408441600af9f7aa105eb9c599b9af12ec50ba868fd82) can include all sorts of dependencies, and we need not be as strict and only list the required dependencies.

We can keep `pymongo` here, and add a comment to mention that it is a testing dependency.

```suggestion
  # Python test dependencies
  - pymongo
```

What matters for ArcticDB's distribution on conda-forge is the removal of [`pymongo` as a python runtime dependency on ArcticDB's feedstock recipe](https://github.com/conda-forge/arcticdb-feedstock/blob/main/recipe/meta.yaml#L104).